tests/user_examples/extra_src/shared/wetting_coupled_spatial_temporal_method.h

```diff
@@ -0,0 +1,104 @@
+/* ------------------------------------------------------------------------- *
+ *                                SPHinXsys                                  *
+ * ------------------------------------------------------------------------- *
+ * SPHinXsys (pronunciation: s'finksis) is an acronym from Smoothed Particle *
+ * Hydrodynamics for industrial compleX systems. It provides C++ APIs for    *
+ * physical accurate simulation and aims to model coupled industrial dynamic *
+ * systems including fluid, solid, multi-body dynamics and beyond with SPH   *
+ * (smoothed particle hydrodynamics), a meshless computational method using  *
+ * particle discretization.                                                  *
+ *                                                                           *
+ * SPHinXsys is partially funded by German Research Foundation               *
+ * (Deutsche Forschungsgemeinschaft) DFG HU1527/6-1, HU1527/10-1,            *
+ *  HU1527/12-1 and HU1527/12-4.                                             *
+ *                                                                           *
+ * Portions copyright (c) 2017-2023 Technical University of Munich and       *
+ * the authors' affiliations.                                                *
+ *                                                                           *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may   *
+ * not use this file except in compliance with the License. You may obtain a *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.        *
+ *                                                                           *
+ * ------------------------------------------------------------------------- */
+/**
+ * @file 	fluid_surface_complex.h
+ * @brief 	Here, we define the algorithm classes for fluid dynamics within the body.
+ * @details We consider here weakly compressible fluids. The algorithms may be
+ * 			different for free surface flow and the one without free surface.
+ * @author	Chi Zhang and Xiangyu Hu
+ */
+
+#ifndef WETTING_COUPLED_SPATIAL_TEMPORAL_COMPLEX_H
+#define WETTING_COUPLED_SPATIAL_TEMPORAL_COMPLEX_H
+
+#include "fluid_surface_complex.h"
+
+namespace SPH
+{
+namespace fluid_dynamics
+{
+/**
+ * @class WettingCoupledFreeSurfaceIndicationComplex
+ * @brief indicate the interfacial fluid particles near the fluid-structure intersurface.
+ */
+class FreeSurfaceWettingIndicationComplex : public FreeSurfaceIndicationComplex
```

Comment:
Can we get a better name perhaps? The comment and description is clear but not the name of the class itself, what about `FluidStructureInterfaceIndicator` ? @ChiZhangatTUM

Reply:
In genaral FSI, especially at the fluid-structure interface, we only use the original spatial-temporal identification solely based on the position divergence. This new class is only for water entry, where the wetting need to be considered. The name you suggested is a little general, and this will make guys misunderstand it when they do the general FSI simulations without wetting considered.  In my opion,  we should emphasize the wetting in the class name.

Reply:
ok, I am not familiar enough with the wetting concept. Just reading the class description via the `@brief` misled me, expanding it might be helpful?

Reply:
with your suggestion, I have discussed with Prof, HU. we agree with your suggestion for a better name. 
We name the class as NoWettingSurfaceIndication. I rewrite the discription to explain it.
Here is the new name and the disription, please check it , thank you very much.

 * @class NonWettingSurfaceIndication
 * @brief Non wetting surface particles include free-surface ones and interfacial ones near the non-wetted structure.
 * @brief Even the position divergence of interfacial fluid pariticles has satisfied with the threshold of spatial-temporal 
   identification approach to be identified as internal ones,they will remain as free-surface ones if without 
   any wetted neighboring solid particles.
 */
class NonWettingSurfaceIndication : public FreeSurfaceIndicationComplex